jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SdkSelectionPanel.kt

```diff
@@ -3,45 +3,91 @@
 
 package software.aws.toolkits.jetbrains.services.lambda.wizard
 
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.WriteAction
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.projectRoots.Sdk
+import com.intellij.openapi.roots.ModifiableRootModel
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.openapi.ui.TextFieldWithBrowseButton
 import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.ui.ErrorLabel
+import com.intellij.ui.components.panels.Wrapper
+import com.intellij.ui.layout.panel
+import com.intellij.util.ThrowableRunnable
 import software.amazon.awssdk.services.lambda.model.Runtime
 import software.aws.toolkits.jetbrains.services.lambda.runtimeGroup
 import javax.swing.JComponent
 import javax.swing.JLabel
-import javax.swing.JPanel
 
-// UI for selecting target SDK of a Runtime
-interface SdkSelectionPanel {
-    val sdkSelectionPanel: JComponent
+interface SdkSelector {
+    fun sdkSelectionPanel(): JComponent
 
-    val sdkSelectionLabel: JLabel?
+    fun sdkSelectionLabel(): JLabel?
 
-    fun registerListeners()
+    fun applySdkSettings(model: ModifiableRootModel) {
+        getSdk()?.let { sdk ->
```

Comment:
nit: big block in the `?.let`, `?: return` instead?

Reply:
Changed